src/aes.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use crate::traits::{AuthenticatedCipher, Cipher, EncryptionKey, ToFromBytes};
+
+use aes::cipher::{BlockDecryptMut, BlockEncryptMut};
+use aes_gcm::{AeadInPlace, KeyInit};
+use ctr::cipher::StreamCipher;
+use digest::{crypto_common::KeyIvInit, generic_array::ArrayLength, typenum::U16};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+///
+/// An encryption key for a symmetric cipher consisting of `N` bytes.
+///
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Key<const N: usize> {
+    // This is needed to derive Serialize and Deserialize since serde does not support derive for arrays
```

Comment:
cc: @huitseeker to sign off the idiomatic way

Reply:
Yep, this is good.